### PR TITLE
List items action

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: application/grifts/db.go [hunk 1]
```diff
@@ -4,88 +4,279 @@ import (
...
+	"github.com/silinternational/riskman-api/api"
+
 	"github.com/gofrs/uuid"
 	"github.com/markbates/grift/grift"
 
 	"github.com/silinternational/riskman-api/domain"
 	"github.com/silinternational/riskman-api/models"
 )
 
+func stringToUUID(input string) uuid.UUID {
```
Comment (Schparky) [inline application/grifts/db.go:16]: I think this is the same as `uuid.FromStringOrNil`

File: application/models/item.go [hunk 2]
```diff
@@ -102,3 +96,57 @@ func (i *Item) LoadPolicy(tx *pop.Connection, reload bool) error {
 
 	return nil
 }
+
+// LoadCategory - a simple wrapper method for loading an item category on the struct
+func (i *Item) LoadCategory(tx *pop.Connection, reload bool) error {
+	if i.Category.ID == uuid.Nil || reload {
+		if err := tx.Load(i, "Category"); err != nil {
+			return err
```
Comment (Schparky) [inline application/models/item.go:104]: I think this can panic since no user-level errors can happen on Load.

**Review by Schparky — APPROVED** [review]

File: application/models/item.go [hunk 3]
```diff
@@ -102,3 +96,57 @@ func (i *Item) LoadPolicy(tx *pop.Connection, reload bool) error {
...
+		}
+	}
+
+	return nil
+}
+
+func ConvertItem(tx *pop.Connection, item Item) (api.Item, error) {
```
Comment (jason-jackson) [inline application/models/item.go:111]: I think we're switching this to only return converted item, as error will always be nil.

File: application/models/item.go [hunk 4]
```diff
@@ -102,3 +96,57 @@ func (i *Item) LoadPolicy(tx *pop.Connection, reload bool) error {
...
+		CoverageAmount:    item.CoverageAmount,
+		PurchaseDate:      item.PurchaseDate,
+		CoverageStatus:    item.CoverageStatus,
+		CoverageStartDate: item.CoverageStartDate,
+		CreatedAt:         item.CreatedAt,
+		UpdatedAt:         item.UpdatedAt,
+	}, nil
+}
+
+func ConvertItems(tx *pop.Connection, items Items) (api.Items, error) {
```
Comment (jason-jackson) [inline application/models/item.go:141]: I think we're switching this to only return converted item, as error will always be nil.

File: application/models/riskcategories.go [hunk 5]
```diff
@@ -45,3 +47,26 @@ func RiskCategoryMobileID() uuid.UUID {
 func RiskCategoryStationaryID() uuid.UUID {
 	return uuid.FromStringOrNil(RiskCategoryStationaryIDString)
 }
+
+func ConvertRiskCategory(rCat RiskCategory) (api.RiskCategory, error) {
```
Comment (jason-jackson) [inline application/models/riskcategories.go:51]: I think we're switching this to only return converted item, as error will always be nil.

File: application/models/riskcategories.go [hunk 6]
```diff
@@ -45,3 +47,26 @@ func RiskCategoryMobileID() uuid.UUID {
...
+	return api.RiskCategory{
+		ID:        rCat.ID,
+		Name:      rCat.Name,
+		PolicyMax: rCat.PolicyMax,
+		CreatedAt: rCat.CreatedAt,
+		UpdatedAt: rCat.UpdatedAt,
+	}, nil
+}
+
+func ConvertRiskCategories(iCats RiskCategories) (api.RiskCategories, error) {
```
Comment (jason-jackson) [inline application/models/riskcategories.go:61]: I think we're switching this to only return converted item, as error will always be nil.

File: application/actions/dependents_test.go [hunk 7]
```diff
@@ -121,6 +121,11 @@ func (as *ActionSuite) Test_DependentsCreate() {
 		BirthYear: 1999,
 	}
 
+	goodRequest2 := api.PolicyDependentInput{
+		Name:      "dependent name2",
+		BirthYear: 2001,
```
Comment (jason-jackson) [inline application/actions/dependents_test.go:126]: Will change in my PR.

**Review by jason-jackson — APPROVED** [review]

File: application/actions/items.go [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+	policy := getReferencedPolicyFromCtx(c)
+	if policy == nil {
+		err := errors.New("policy not found in context")
+		return reportError(c, api.NewAppError(err, api.ErrorGettingPolicyFromContext, api.CategoryInternal))
+
+	}
+
+	err := policy.LoadItems(tx, true)
+	if err != nil {
+		return c.Render(http.StatusInternalServerError, r.JSON(err))
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/actions/items.go:24]: :+1:

File: application/actions/items_test.go [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+		as.T().Run(tt.name, func(t *testing.T) {
+			req := as.JSON("/policies/%s/items", tt.policy.ID.String())
+			req.Headers["Authorization"] = fmt.Sprintf("Bearer %s", tt.actor.Email)
+			req.Headers["content-type"] = "application/json"
+			res := req.Get()
+
+			body := res.Body.String()
+			as.Equal(tt.wantStatus, res.Code, "incorrect status code returned, body: %s", body)
+
+			for _, w := range tt.wantInBody {
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/actions/items_test.go:93]: :+1:

File: application/grifts/db.go (hunk 1 above)
Comment (Baggerone) [reply application/grifts/db.go:16]: :+1:

File: application/models/item.go (hunk 2 above)
Comment (Baggerone) [reply application/models/item.go:104]: :+1:

**Pull request merged**